projects/graphql-client/src/model/graphql-selection.ts

```diff
@@ -2,6 +2,7 @@ import { GraphQlArgument } from './graphql-argument';
 
 export interface GraphQlSelection {
   path: string;
+  name?: string;
```

Comment:
how is this different than alias?

Reply:
My guess from the other code is this is used by the gql client to pass metadata from the request options to the root selection, but this type is used for general request building so we don't want to leak an internal impl detail into it.

Reply:
Yes...I think I could hide it with a new interface which is not exposed.

Reply:
It's very similar to what you're doing - it provides a custom name for a field (so not usually used at root). Most commonly, we see this when the same field is queried multiple times with different arguments.

e.g. an entity requesting
```
  name: attribute(key: "name")
  otherAttr: attribute(key: "otherAttr")
```